### DISCUSSION TITLE
SSL decryption

### DISCUSSION THREAD

**Discussion by Jackson-Pollock** [Security-Onion-Solutions/securityonion#6483] [Unsupported Versions] [answered]
As I understood Suricata alone can not handle encrypted packet inspection and the reason why my reverse_https shell is undetected. I want to understand how Suricata can inspect HTTPS connections and then malicious traffic can be detected?

**Comment by Jackson-Pollock** [comment, marked as answer]
Is there any provision in security onion or we need to rely on 3rd party proxies/decrypter.

**Reply by presianbg** [reply]
Hi, 

There is no way to decrypt the SSL\TLS traffic on a IDS system, since its not terminating the sessions therefore you can't get the SSL\TLS client session keys. 
You may want to check this great talk - https://www.youtube.com/watch?v=-WD9BWlENwc

Yes, your best choice is to have some in-line proxy for decrypting and insepcting the SSL\ TLS traffic, then forward the logs to SO or any other SIEM-ish solution.

Cheers,
PY

**Reply by presianbg** [reply]
Also you may check #3566

**Reply by Jackson-Pollock** [reply]
Thank you @presianbg

**Comment by UMHB-InfoSec** [comment]
If I may resurrect this thread.

I am looking at something similar by using squid to act as a transparent SSL proxy.
If squid is proxy-ing the connections, on the inside the traffic will be encrypted using a cert that I control.

With that being the case would it be possible to have SO do some kind of auto-decrypting using my cert info so that my pcaps would show me the decrypted info?

**Reply by presianbg** [reply]
No, having the CA used on the proxy, does not make you "knowing" the session keys needed for successful SSL\TLS inspection.
Check this out:
https://www.netresec.com/?page=Blog&month=2020-01&post=Sniffing-Decrypted-TLS-Traffic-with-Security-Onion

In essence you have to `tcpreplay` the plain HTTP traffic to SO monitoring interface. 
Some of the pricey NGFW vendors are having an option to transparently intercept the HTTPS (you have to trust the appliance CA on all OS / Apps)  and send a plain text copy of the encrypted traffic to somewhere (SIEM, Log Collector, SO, etc...).

With all this said, I have my doubts all this pain and $$$ will ever pay off.

Let me know if you have more questions. 

Cheers,
PY

**Reply by UMHB-InfoSec** [reply]
Dagnabbit, I see what you mean.

Looks like ja3 is the way of the future I guess.

Please excuse my ignorance : )
